src/FakeClient/FakeClientMain.cpp

```diff
@@ -57,6 +61,40 @@ void InstallSigintHandler() {
   sigaction(SIGINT, &act, nullptr);
 }
 
+// On OrbitService's side, and in particular in LinuxTracing, the only information needed to
+// instrument a function is what uprobes need, i.e., the path of the module and the function's
+// offset in the module (address minus load bias). But CaptureClient requires much more than that,
+// through the ModuleManager and the FunctionInfos. For now just keep it simple and leave the fields
+// that are not needed empty.
+void ManipulateModuleManagerAndSelectedFunctionsToAddUprobe(
+    orbit_client_data::ModuleManager* module_manager,
+    absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo>* selected_functions,
+    const std::string& file_path, uint64_t file_offset) {
+  ErrorMessageOr<std::unique_ptr<orbit_elf_utils::ElfFile>> error_or_elf_file =
+      orbit_elf_utils::ElfFile::Create(std::filesystem::path{file_path});
+  FAIL_IF(error_or_elf_file.has_error(), "%s", error_or_elf_file.error().message());
+  std::unique_ptr<orbit_elf_utils::ElfFile>& elf_file = error_or_elf_file.value();
+  std::string build_id = elf_file->GetBuildId();
+  ErrorMessageOr<uint64_t> error_or_load_bias = elf_file->GetLoadBias();
+  FAIL_IF(error_or_load_bias.has_error(), "%s", error_or_load_bias.error().message());
+  uint64_t load_bias = error_or_load_bias.value();
+
+  orbit_grpc_protos::ModuleInfo module_info;
+  module_info.set_file_path(file_path);
+  module_info.set_build_id(build_id);
+  module_info.set_load_bias(load_bias);
+  CHECK(module_manager->AddOrUpdateModules({module_info}).empty());
+
+  orbit_client_protos::FunctionInfo function_info;
+  function_info.set_module_path(file_path);
+  function_info.set_module_build_id(build_id);
+  function_info.set_address(load_bias + file_offset);
```

Comment:
Should we pass the address instead of the offset as command line argument? What do you think?